tests/testthat/test-score_forecasts.R

```diff
@@ -2456,3 +2456,403 @@ test_that("wis is sum of components, 2 intervals, median and point, use_median_a
+    value = y,
+    stringsAsFactors = FALSE
+  )
+  
+  n_forecasts <- length(forecast_quantiles)
+  test_forecasts <- data.frame(
+    model = rep("m1", n_forecasts),
+    forecast_date = rep(lubridate::ymd("20200101"), n_forecasts),
+    location = forecast_locations,
+    horizon = forecast_horizons,
+    temporal_resolution = rep("wk", n_forecasts),
+    target_variable = forecast_target_variables,
+    target_end_date = forecast_target_end_dates,
+    type = rep("quantile", n_forecasts),
+    quantile = forecast_quantile_probs,
+    value = forecast_quantiles,
+    stringsAsFactors = FALSE
+  )
+  
+  actual <- score_forecasts(forecasts = test_forecasts, truth = test_truth, 
+                            use_median_as_point = TRUE)
+  
+  
+  expected_coverage_50 <- ifelse(y>=forecast_quantiles_matrix[, 1] & y <= forecast_quantiles_matrix[, 2], 1, 0)
+  
+  expect_equal(actual$coverage_50, expected_coverage_50)
+})
+
+test_that("interval coverage  is correct, 2 intervals and median no point, use_median_as_point TRUE", {
+  y <- c(1, -1, 5)
+  forecast_quantiles_matrix <- rbind(
+    c(-1, 0, 1, 2, 3),
+    c(-2, -1, 2, 2, 4),
+    c(-2, 0, 3, 4, 6))
+  forecast_quantile_probs <- c(0.1, 0.25, 0.5, 0.75, 0.9)
+  
+  target_end_dates <- lubridate::ymd(20200101) + c(7, 14, 7)
+  horizons <- c("1", "2", "1")
+  locations <- c("01", "01", "02")
+  target_variables <- rep("inc death", length(y))
+  
+  forecast_target_end_dates <-
+    rep(target_end_dates, times = ncol(forecast_quantiles_matrix))
+  forecast_horizons <- rep(horizons, times = ncol(forecast_quantiles_matrix))
+  forecast_locations <- rep(locations, times = ncol(forecast_quantiles_matrix))
+  forecast_target_variables <-
+    rep(target_variables, times = ncol(forecast_quantiles_matrix))
+  forecast_quantile_probs <- rep(forecast_quantile_probs, each = length(y))
+  forecast_quantiles <- forecast_quantiles_matrix
+  dim(forecast_quantiles) <- prod(dim(forecast_quantiles))
+  
+  test_truth <- data.frame(
+    model = rep("truth_source", length(y)),
+    target_variable = target_variables,
+    target_end_date = target_end_dates,
+    location = locations,
+    value = y,
+    stringsAsFactors = FALSE
+  )
+  
+  n_forecasts <- length(forecast_quantiles)
+  test_forecasts <- data.frame(
+    model = rep("m1", n_forecasts),
+    forecast_date = rep(lubridate::ymd("20200101"), n_forecasts),
+    location = forecast_locations,
+    horizon = forecast_horizons,
+    temporal_resolution = rep("wk", n_forecasts),
+    target_variable = forecast_target_variables,
+    target_end_date = forecast_target_end_dates,
+    type = rep("quantile", n_forecasts),
+    quantile = forecast_quantile_probs,
+    value = forecast_quantiles,
+    stringsAsFactors = FALSE
+  )
+  
+  actual <- score_forecasts(forecasts = test_forecasts, truth = test_truth, 
+                            use_median_as_point = TRUE)
+  
+  expected_coverage_0 <- ifelse(y==forecast_quantiles_matrix[, 3], 1, 0)
```

Comment:
again, I think we should remove the coverage_0 as it doesn't make sense as an output.